File: src/types/_AlternativeNames.ts
import * as DB from "tsondb/schema/dsl"

export const AlternativeName = DB.TypeAlias(import.meta.url, {
  name: "AlternativeName",
  type: () =>
    DB.Object({
      name: DB.Required({
        comment: "An alternative name of the disease.",
        type: DB.String({ minLength: 1 }),
      }),
      region: DB.Optional({
        comment: "The region where this alternative name is used.",
        type: DB.String({ minLength: 1 }),
      }),
    }),
})

File: Sources/OptolithDatabaseSchema/GeneratedTypes/_AlternativeNames.swift
//
//  _AlternativeNames.swift
//  OptolithDatabaseSchema
//

import DiscriminatedEnum

public struct AlternativeName: EntitySubtype {
    /// An alternative name of the disease.
    public let name: NonEmptyString
    
    /// The region where this alternative name is used.
    public let region: NonEmptyString?

    public init(name: NonEmptyString, region: NonEmptyString? = nil) {
        self.name = name
        self.region = region
    }
}
